src/middlewared/middlewared/plugins/system/utils.py

```diff
@@ -23,8 +24,10 @@ def __init__(self):
         self.SYSTEM_SHUTTING_DOWN = False
 
 
-def get_debug_execution_dir(system_dataset_path: str) -> str:
-    return '/var/tmp/ixdiagnose' if system_dataset_path is None else os.path.join(system_dataset_path, 'ixdiagnose')
+def get_debug_execution_dir(system_dataset_path: str, iteration: typing.Optional[int] = 0) -> str:
+    return f'/var/tmp/ixdiagnose-{iteration}' if system_dataset_path is None else os.path.join(
```

Comment:
I do agree with @anodos325 's comment about putting this in `/tmp`. We're trying to not perform operations within that filesystem for obvious reasons. We have a middleware run directory that is somewhat more secure than `/tmp` that we can put these in.